docs/dashboard.md

```diff
@@ -150,6 +150,16 @@ The `@csv.websites.url` syntax will look up a `Source` called 'csv',
 request a table called 'websites' and then feed the 'url' column in
 that table to the `urls` parameter of the `WebsiteSource`.
 
+### Templating
+
+In many cases you do not want to hardcode variables inside the yaml specification instead passing in variables from an environment variable, a shell command or as a CLI argument. This can be achieved using templating syntax:
+
+- `{{env("USER")}}`: look in the set environment variables for the named variable
+
+- `{{shell("get_login thisuser -t")}}`: execute the command, and use the output as the value. The output will be trimmed of any trailing whitespace.
```

Comment:
This looks very useful, but do we now need to offer a safe_load option for the YAML that disables features like this, so that people could safely write a web app for dynamically constructing Lumen apps from an editable (and thus abusable) YAML spec?

Reply:
Yes, especially during live edits I should not allow this.

Reply:
Hmm, during live edits is access to environment variables safe? Seems like I should not allow any of these during live editing.

Reply:
How would you suggest we toggle this outside of live editing? A commandline argument?

Reply:
Will merge for now, but I agree this is a very valid concern and we have to consider how to communicate this and toggling the behavior.

Reply:
Yes, I'm assuming a commandline argument, because whether such support is appropriate is about where you are launching the server more than about the contents of the dashboard.

Reply:
But I think it should also be available from the Python API interface to lumen, so that a Panel app that allows .yml to be edited can call lumen in safe mode to render it.